backtop/backtop.js

```diff
@@ -21,30 +43,40 @@ var Backtop = React.createClass({
 				isShow: false
 			})
 		}
-	},
-	backToTop() {
-		$("html, body").animate({scrollTop: 0}, "easeInOut");
-	},
-	componentDidMount() {
-		$(window).on("scroll", this.onScroll);
-	},
-	componentWillUnmount() {
-		$(window).off("scroll", this.onScroll);
-	},
-	buildComponent(props, state) {
-		let _className = cx({
-			"b-backtop": true,
-			[this.props.className]: !!props.className
+	}
+
+	backToTop = () => {
+		// http://stackoverflow.com/questions/21474678/scrolltop-animation-without-jquery
+
+		const scrollDuration = this.props.scrollDuration;
+		let cosParameter = window.scrollY / 2,
+			scrollCount = 0,
+			oldTimestamp = performance.now();
+		function step (newTimestamp) {
```

Comment:
move this function outside, otherwise you're creating a new function everytime backtop called